Source/WebCore/dom/Document.cpp

```diff
@@ -7385,21 +7409,21 @@ Document* Document::parentDocument() const
     return parent->document();
 }
 
-Document& Document::topDocument() const
+Document* Document::mainFrameDocument() const
 {
     // FIXME: This special-casing avoids incorrectly determined top documents during the process
     // of AXObjectCache teardown or notification posting for cached or being-destroyed documents.
     if (backForwardCacheState() == NotInBackForwardCache && !m_renderTreeBeingDestroyed) {
         Document* localMainDocument = nullptr;
         if (RefPtr localMainFrame = this->localMainFrame())
             localMainDocument = localMainFrame->document();
-        return localMainDocument ? *localMainDocument : const_cast<Document&>(*this);
+        return localMainDocument ? localMainDocument : const_cast<Document*>(this);
```

Comment:
We may eventually want a "const Document* mainFrameDocument()" const and "Document* mainFrameDocument()" but definitely not in this PR.